pkg/config/namewatcher.go

```diff
@@ -0,0 +1,174 @@
+package config
+
+import (
+	"context"
+	"strings"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+
+	"github.com/openshift/csi-driver-shared-resource/pkg/client"
+)
+
+const (
+	sharedSecretReservedNamesConfigMapName    = "ocp-sharedsecret-list"
+	sharedConfigMapReservedNamesConfigMapName = "ocp-sharedconfigmap-list"
+)
+
+type ReservedNames struct {
+	sharedSecretReservedNamesWatch    watch.Interface
+	sharedConfigMapReservedNamesWatch watch.Interface
+	currentSecretNames                map[string]string
+	currentConfigMapNames             map[string]string
+	lock                              sync.Mutex
+}
+
+func SetupNameReservation( /*stopCh <-chan struct{}*/ ) *ReservedNames { //}, error) {
+	ctx := context.Background()
+	rn := &ReservedNames{
+		lock: sync.Mutex{},
+	}
+	var err error
+	//rn.sharedSecretReservedNamesWatch, err = client.GetClient().CoreV1().ConfigMaps(client.DefaultNamespace).Watch(ctx, metav1.ListOptions{FieldSelector: "metadata.name=" + sharedSecretReservedNamesConfigMapName})
+	//if err != nil {
+	//	return nil, err
+	//}
+	//rn.sharedConfigMapReservedNamesWatch, err = client.GetClient().CoreV1().ConfigMaps(client.DefaultNamespace).Watch(ctx, metav1.ListOptions{FieldSelector: "metadata.name=" + sharedConfigMapReservedNamesConfigMapName})
+	//if err != nil {
+	//	return nil, err
+	//}
+	var sNames, cmNames *corev1.ConfigMap
+	sNames, err = client.GetClient().CoreV1().ConfigMaps(client.DefaultNamespace).Get(ctx, sharedSecretReservedNamesConfigMapName, metav1.GetOptions{})
+	if err == nil {
+		rn.updateSecretNames(sNames.Data)
+	}
+	cmNames, err = client.GetClient().CoreV1().ConfigMaps(client.DefaultNamespace).Get(ctx, sharedConfigMapReservedNamesConfigMapName, metav1.GetOptions{})
+	if err == nil {
+		rn.updateConfigMapNames(cmNames.Data)
+	}
+	//go rn.watchReservedNamesConfigMap(ctx, stopCh, rn.sharedConfigMapReservedNamesWatch)
```

Comment:
I'll clean this up once e2e's look OK. Don't think I need it but wanted to wait and make sure.